src/core/Value.cc

```diff
@@ -328,6 +329,10 @@ class tostream_visitor : public boost::static_visitor<>
   }
 
   void operator()(const VectorType& v) const {
+    if (StackCheck::inst().check()) { 
+      LOG(message_group::Error, Location::NONE, "", "Stack exhausted while trying to convert a vector to EchoString");
```

Comment:
The LOG should not be done immediately since, particularly in Qt GUI, this process can add a lot to the stack.  It should be deferred to somewhere up the stack after catching the exception somewhere.

I'm also wondering where the following exception ends up getting caught, and wouldn't that cause a duplicated message?

I would suggest to just throw the exception here, without the LOG call.  
Then try/catch for it inside the **`tostring_visitor`**'s  `std::string operator()(const VectorType& v) const`, with the LOG call after catching it.

Basically `tostream_visitor` only is called through `tostring_visitor`'s vector handling function, so catching this will ensure that the exception will pop off the stack a good amount (to the top level) before attempting to print the log message.

Reply:
I will respond in detail.
The response in short is, that I used `RecursionException` as a design pattern.